src/components/Graph.react.js

```diff
@@ -109,6 +109,18 @@ class PlotlyGraph extends Component {
         );
     }
 
+    extend(props) {
+        const {id, extendData} = props;
+        const gd = document.getElementById(id);
+        const [updateData, traceIndices] = extendData;
+
+        if (updateData && gd.data.length >= 1) {
+            return Plotly.extendTraces(id, updateData, traceIndices);
+        }
+
+        return this.plot(props);
```

Comment:
In `componentWillReceiveProps` we've already done `this.plot` if appropriate prior to `this.extend`. Do we still want this one?

Reply:
I prefer to keep this in place. In the case that the Graph is initialized with no data (`gd.data.length === 0`, the component will throw an error because of invalid values in traceIndices.

Reply:
That's why you have the `gd.data.length >= 1` above, but I don't see what that has to do with calling `this.plot` as a fallback inside of `extend`. Perhaps we need a test case covering the situation you're concerned about?